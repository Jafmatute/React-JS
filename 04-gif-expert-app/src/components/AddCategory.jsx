import {useState} from "react";

const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('One punch');


    const onInputChange = ({target:{value}}) => setInputValue(value);

    const onSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length <=1) return;

        setCategories(category => [...category, inputValue]);

        setInputValue('');
    }


    return (
        <form onSubmit={onSubmit}>
        <input
            type='text'
            placeholder="Buscar gits"
            value={inputValue}
            onChange={onInputChange} />
        </form>
    );
};

export default AddCategory;
