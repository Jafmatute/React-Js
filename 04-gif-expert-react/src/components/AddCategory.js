/** @format */

import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategorias }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChanged = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategorias((cats) => [inputValue, ...cats]);
      setInputValue("");
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChanged} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategorias: PropTypes.func.isRequired,
};

export default AddCategory;
