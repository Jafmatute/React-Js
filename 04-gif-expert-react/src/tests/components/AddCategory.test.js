import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas en el componeten <AddCategory />', () => {
  const setCategorias = jest.fn();
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategorias={setCategorias} />);
  });

  test('Verificar el componente se renderice', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo';
    input.simulate('change', { target: { value } });

    //   expect(wrapper.find('p').text().trim()).toBe(value);
  });

  test('No debe de postear la informaciÃ³n submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategorias).not.toHaveBeenCalled();
    //si deseo se hubiese llamado
    //expect(setCategorias).toHaveBeenCalled();
  });
});
