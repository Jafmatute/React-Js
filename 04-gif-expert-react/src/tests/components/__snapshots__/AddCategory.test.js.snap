// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componeten <AddCategory /> Verificar el componente se renderice 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
